### ISSUE TITLE
Cannot handle addresses with many transactions

### ISSUE THREAD

**Issue by trevinhofmann** [thofmann/mychain#27]
Heavily reused addresses cannot be displayed, as it requires too many transactions to be retrieved from the RPC. Perhaps it should display the amount received from the first X transactions, and add a '+' to the number with a message describing the problem?

**Comment by alexanderkent** [comment]
http://mychain.io/address/1QB4wdHKAX1k8PwDi2G4pqB7SWMy4JSdsZ

**Comment by alexanderkent** [comment]
And try this one 1dice8EMZmqKvrGE4Qc9bUFf9PX3xaYDp  -- no dice

**Comment by trevinhofmann** [comment]
Hey, thanks for the heads up!

The first address did load for me, but took 19.05 seconds before giving up and sending the transactions it had so far. Maybe I should lower the threshold to give up sooner. I also still need to add the message alerting the user of the problem.

**Comment by trevinhofmann** [comment]
I am not getting any response for the satoshi dice address.

**Comment by alexanderkent** [comment]
Yup 3.1 million tx for the satoshi dice address... Top of my head it would need some sort of pagination system though for that it would probably need a refactered data tier. I just stumbled over your repo so I am not familiar with it, are you using mysql? Did you consider mongodb / redis?

**Comment by trevinhofmann** [comment]
Yes, this was meant to be a fairly lightweight explorer. I will probably not be keeping the site up, to leave time/funding for other projects.

It is using a MySQL database to index transactions. In the table, each output is stored in a row and the two columns are an indexed hash of the output script and then the txid where it is stored. Then, to look up transactions paying to an address, for example, it builds the redemption script for that address, hashes it, and searches the database. Once it has the txid(s), it can quickly get all of the information from the Bitcoin Core RPC. Compared to other explorers, it is very efficient. Including both the database and Bitcoin Core's block chain, it is still under 100 GB. The database synchronizes on my VPS in just over one day.

The main problem with heavily reused addresses is the time it takes to request and process the transactions from Bitcoin Core. I am not sure if I want to compromise the efficiency to better handle address reuse. Address reuse is a bad practice, and I would rather not accommodate it.
